vignettes/data-repository.Rmd

```diff
@@ -0,0 +1,200 @@
+
+The data file (".tsv") and the meta data file (".yml") are automatically staged after writing. `write_delim_git()` returns the hashes of both the data file and the meta data file. The names of the vector contain the path of the files, relative to the root of the _git_ repository.
+
+```{r}
+status(data_repo)
+write_delim_git(x = x, file = "my_data", repo = data_repo, sorting = c("y", "x"))
+status(data_repo)
+status(repo)
+```
+
+Overwriting data which has different meta data than the original data will throw an error because this can potentially lead to large diffs. Differences in meta data occur when:
+
+- the sorting changes
+- variables are added, removed or renamed
+- the class of a variable changes
+- the levels of a factor changes
+
+However you can bypass this by using `override = TRUE`. This will completely ignore the existing meta data and create new meta data based on the new `data.frame`.
+
+```{r error=TRUE}
+y <- x
+y$extra <- x$x
+write_delim_git(y, file = "my_data", repo = data_repo, sorting = c("y", "x"))
+write_delim_git(
+    y, file = "my_data", repo = data_repo, sorting = c("y", "x"), 
+    override = TRUE
+)
+```
+
+### Reading data
+
+Retrieving data is straight forward. Use `read_delim_git` and provide the `file` and the `repo`. The retrieved data is identical to the original data after applying the ordering of variables and observations.
+
+```{r}
+y_stored <- read_delim_git(file = "my_data", repo = data_repo)
+str(y_stored)
+y_sorted <- y[order(y$y, y$x), colnames(y_stored)]
+rownames(y_sorted) <- NULL
+stopifnot(all.equal(y_sorted, y_stored))
+```
+
+### Committing changes
+
+Commit the changes using `commit()`
+
+```{r}
+write_delim_git(x, file = "sub/test", repo = data_repo, sorting = "x")
+status(data_repo)
+commit(data_repo, message = "Initial commit")
+status(data_repo)
+```
+
+### Removing data
+
+Data objects can be removed with `rm_file`. Provide the data repository to `repo` and the data object name to `path`. `path` can take multiple data objects.
+
+```{r}
+# remove a single data object
+rm_file(repo = data_repo, path = "my_data")
+status(data_repo)
+# undo the remove by resetting to the last commit
+reset(commits(data_repo)[[1]], "hard")
+status(data_repo)
+# remove multiple data objects
+rm_file(repo = data_repo, path = c("my_data", "sub/test"))
+status(data_repo)
+```
+
+Note that this will remove both the data and the meta data. In case you want to update the current data, it is better to overwrite the data than first delete it and then write it. Sometime you want to write a bunch of data object to the data repository and you cannot guarantee that all data objects will remain after the update. The workflow below is intended for such case.
```

Comment:
In case you want to update the current data, it is better to overwrite the data than first delete it and then write it ->
In case you want to update the current data, it is better to overwrite the data.

Reply:
SometimeS

Reply:
I also would start a new paragraph or subsection on 'Sometimes...' as you define here a really specific case